projects/packages/videopress/src/class-block-editor-content.php

```diff
@@ -21,31 +21,67 @@ public static function init() {
 			return;
 		}
 
-		add_filter( 'default_content', array( static::class, 'video_block_by_guid' ), 10, 2 );
+		add_shortcode( 'jetpack_videopress', array( static::class, 'videopress_block_shortcode' ) );
+		add_filter( 'default_content', array( static::class, 'videopress_video_block_by_guid' ), 10, 2 );
 	}
 
 	/**
-	 * Generates a video block content with the given guid
+	 * VideoPress embed shortcode
+	 *
+	 * Example use:
+	 * [jetpack_videopress guid=tLvEwHYZ width=560 height=315]
+	 *
+	 * @param array $atts Shortcode attributes.
+	 *
+	 * @return string html
+	 */
+	public static function videopress_block_shortcode( $atts ) {
+		$atts = shortcode_atts(
+			array(
+				'guid'   => '', // string.
+				'width'  => 560, // int.
+				'height' => 315, // int.
+			),
+			$atts,
+			'jetpack_videopress'
+		);
+
+		$guid = sanitize_text_field( wp_unslash( $atts['guid'] ) );
+
+		if ( empty( $guid ) ) {
+			return '<!-- error: missing VideoPress video ID -->';
+		}
+
+		$width  = (int) $atts['width'];
+		$height = (int) $atts['height'];
+		$src    = esc_url( 'https://videopress.com/embed/' . $guid );
+
+		wp_enqueue_script( 'videopress-iframe', 'https://videopress.com/videopress-iframe.js', array(), Package_Version::PACKAGE_VERSION, true );
```

Comment:
the `videopress-iframe.js` is always loaded. Adding it here makes the browser load it one more time. Not sure what the conditions are to load the script.

Reply:
It is not loaded. It is added as part of the embed code given by the player.
ref p1670448317436549/1670354139.628999-slack-C03TA48NSUX

Without it the player does not load for private videos.